docs/INTEGRATION.md

```diff
@@ -27,7 +26,334 @@ The bundle should be included in the `<head>` tag, ideally near the top of the n
 
 ```
 
-The "referrer" `meta` tag is necessary for enabling branch overrides in the URL (?branch=branchname). This functionality is only currently possible with Searchspring maintained Snap repositories.
+## Context variables
+
+Context variables are conditionally rendered within the script's innerHTML. They provide context for Snap and can be retrieved using [getContext](https://github.com/searchspring/snap/tree/main/packages/snap-toolbox/src/getContext)
+
+The innerHTML of the script MUST only contain variable assignments without `var`, `let`, or `const`. Each declaration should end with a semi-colon to ensure minification does not impact the functions ability to parse the innerHTML.
+
+```html
+<script src="https://snapui.searchspring.io/[your_site_id]/bundle.js">
+	product = 'C-AD-W1-1869P';
+	shopper = {
+		id: 'snapdev',
+		cart: [
+			{ sku: 'product_1' }
+		]
+	};
+	options = {
+		siteId: 'abc123',
+		categories: ['righteous', 'awesome', 'radical']
+	};
+</script>
+```
+
+| Option | Value | Page | Description |
+|---|---|:---:|---|
+| product | current product sku | product detail page | required if product detail pages contain recommendations |
+| shopper.id | logged in user unique identifier | all | required for personalization functionallity |
+| shopper.cart | current cart contents | all | required if checkout process does not contain a dedicated cart page (ie. slideout cart) |
+| options.siteId | global siteId overwrite | all | optional global siteId overwrite |
+| options.categories | category path | recommendations | optional category path |
+| branch | template branch overwrite | recommendations | optional branch overwrite for recommendations template |
+| batched | boolean (default: `true`) | recommendations | only applies to recommendation context, optional disable profile from being batched in a single request, can also be set globally [via config]((https://github.com/searchspring/snap/tree/main/packages/snap-controller/src/Recommendation)) |
+
```

Comment:
It may be a bit confusing grouping the context options in this way... It may be less confusing to split these out into integration context `searchspring-context` and recommendations context - as they have different options and are specified in different scripts blocks.

Integration context:
```
<script src="https://snapui.searchspring.io/[your_site_id]/bundle.js">
  context = 'goes here';
</script>
```

^ for this context the only required variables are `shopper.id` - `shopper.cart` is beneficial as pointed out on sites without a dedicated cart page.

Do we already have a section for recommendations script blocks and the options to be used there? If so, it may be better to leave that to its own doc section.